### DISCUSSION TITLE
x-hasura-allowed-roles not forwarded to remote schema in hasura 2?

### DISCUSSION THREAD

**Discussion by ManUtopiK** [hasura/graphql-engine#6659] [Question]
Is there any reason `x-hasura-allowed-roles` are not forwarded to remote schema in hasura 2?

**Comment by s1monj** [comment]
@ManUtopiK were you able to find any workaround for this?

**Reply by ManUtopiK** [reply]
In my server which provide the remote schema, I take the user ID in the Headers from Hasura, and I re-query Hasura to get allowed roles. This is not optimal, but I didn't find another solution to get users allowed roles...

**Reply by s1monj** [reply]
Gotcha thanks! How are you re-querying Hasura to extract the allowed roles? I combed through the [API docs](https://hasura.io/docs/latest/graphql/core/api-reference/index.html) but couldn't find anything

**Reply by ManUtopiK** [reply]
I store allowed roles in the `user` table in a `jsonb` format. So I simply query Hasura like this:
```
query User($id: uuid!) {
  user_by_pk(id: $id) {
    roles
  }
}
```
